baseapp/baseapp.go

```diff
@@ -726,6 +731,9 @@ func (app *BaseApp) runTx(mode runTxMode, txBytes []byte, tx sdk.Tx) (result sdk
 		if mode == runTxModeDeliver {
 			ctx.BlockGasMeter().ConsumeGas(
 				ctx.GasMeter().GasConsumedToLimit(), "block gas meter")
+			if ctx.BlockGasMeter().GasConsumed() < startingGas {
+				panic(sdk.ErrorGasOverflow{"tx gas summation"})
```

Comment:
Darn, I think we should use this for the error we return in `StdTx#ValidateBasic` too. If you agree, can you change that too?